atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1302,16 +1299,33 @@ private int estimateSize(List<RowResult<Value>> rangeRows) {
             timestampCandidates = keyValueService.getLatestTimestamps(table, nextTimestampCandidates);
         }
 
+        deleteOrphanedSentinelsAsync(table, actualOrphanedSentinels);
+
         return actualOrphanedSentinels;
     }
 
+    private void deleteOrphanedSentinelsAsync(TableReference table, Set<Cell> actualOrphanedSentinels) {
+        sweepQueue.getSweepStrategy(table).ifPresent(strategy -> {
+            if (strategy == SweeperStrategy.THOROUGH) {
```

Comment:
note: it's important that THOROUGH_MIGRATION _does not_ choose the THOROUGH sweep strategy (which it doesn't)